diff --git a/third_party/lcms/src/lcms2_internal.h b/third_party/lcms/src/lcms2_internal.h
index 727cb06e3..115ac5970 100644
--- a/third_party/lcms/src/lcms2_internal.h
+++ b/third_party/lcms/src/lcms2_internal.h
@@ -28,7 +28,7 @@
 
 // Include plug-in foundation
 #ifndef _lcms_plugin_H
-#   include "lcms2_plugin.h"
+#   include "third_party/lcms/include/lcms2_plugin.h"
 #endif
 
 // ctype is part of C99 as per 7.1.2
