diff --git a/third_party/libopenjpeg20/j2k.c b/third_party/libopenjpeg20/j2k.c
index d95963a5c..ed449684f 100644
--- a/third_party/libopenjpeg20/j2k.c
+++ b/third_party/libopenjpeg20/j2k.c
@@ -9864,7 +9864,9 @@ static OPJ_BOOL opj_j2k_read_SQcd_SQcc(opj_j2k_t *p_j2k,
             p_j2k->m_specific_param.m_decoder.m_default_tcp;
 
     /* precondition again*/
-    assert(p_comp_no <  p_j2k->m_private_image->numcomps);
+    if (p_comp_no >=  p_j2k->m_private_image->numcomps) {
+        return OPJ_FALSE;
+    }
 
     l_tccp = &l_tcp->tccps[p_comp_no];
     l_current_ptr = p_header_data;
