diff --git a/third_party/agg23/agg_path_storage.h b/third_party/agg23/agg_path_storage.h
index dc13851..7f21bac 100644
--- a/third_party/agg23/agg_path_storage.h
+++ b/third_party/agg23/agg_path_storage.h
@@ -38,9 +38,9 @@ public:
         }
         unsigned vertex(float* x, float* y)
         {
-            return (m_vertex_idx < m_path->total_vertices()) ?
-                   m_path->vertex(m_vertex_idx++, x, y) :
-                   path_cmd_stop;
+          return (m_vertex_idx < m_path->total_vertices())
+                     ? m_path->vertex(m_vertex_idx++, x, y)
+                     : static_cast<unsigned>(path_cmd_stop);
         }
     private:
         const path_storage* m_path;
